micrometer-core/src/main/java/io/micrometer/core/instrument/binder/hystrix/MicrometerMetricsPublisherThreadPool.java

```diff
@@ -0,0 +1,142 @@
+    this.meterRegistry = meterRegistry;
+    this.threadPoolKey = threadPoolKey;
+    this.metrics = metrics;
+    this.properties = properties;
+    this.metricsPublisherForThreadPool = metricsPublisherForThreadPool;
+
+    this.tags = Tags.of("key", threadPoolKey.name());
+  }
+
+  @Override
+  public void initialize() {
+    metricsPublisherForThreadPool.initialize();
+
+    Gauge.builder(metricName("threads.active.current.count"), metrics::getCurrentActiveCount)
+        .description("The approximate number of threads that are actively executing tasks.")
+        .tags(tags)
+        .register(meterRegistry);
+
+    Gauge.builder(metricName("threads.active.rolling.max"), metrics::getRollingMaxActiveThreads)
+        .description("Rolling max number of active threads during rolling statistical window.")
+        .tags(tags)
+        .register(meterRegistry);
+
+    Gauge.builder(metricName("threads.executed.cumulative.count"), metrics::getCumulativeCountThreadsExecuted)
+        .description("Cumulative count of number of threads executed since the start of the application.")
+        .tags(tags)
+        .register(meterRegistry);
+
+    Gauge.builder(metricName("threads.rejected.cumulative.count"), metrics::getCumulativeCountThreadsRejected)
+        .description("Cumulative count of number of threads rejected since the start of the application.")
+        .tags(tags)
+        .register(meterRegistry);
+
+    Gauge.builder(metricName("threads.executed.rolling.count"), metrics::getRollingCountThreadsExecuted)
+        .description("Rolling count of number of threads executed during rolling statistical window.")
+        .tags(tags)
+        .register(meterRegistry);
+
+    Gauge.builder(metricName("threads.rejected.rolling.count"), metrics::getRollingCountThreadsRejected)
+        .description("Rolling count of number of threads rejected during rolling statistical window.")
+        .tags(tags)
+        .register(meterRegistry);
+
+    Gauge.builder(metricName("threads.pool.current.size"), metrics::getCurrentPoolSize)
+        .description("The current number of threads in the pool.")
+        .tags(tags)
+        .register(meterRegistry);
+
+    Gauge.builder(metricName("threads.largest.pool.current.size"), metrics::getCurrentLargestPoolSize)
+        .description("The largest number of threads that have ever simultaneously been in the pool.")
+        .tags(tags)
+        .register(meterRegistry);
+
+    Gauge.builder(metricName("threads.max.pool.current.size"), metrics::getCurrentMaximumPoolSize)
+        .description("The maximum allowed number of threads.")
+        .tags(tags)
+        .register(meterRegistry);
+
+    Gauge.builder(metricName("threads.core.pool.current.size"), metrics::getCurrentCorePoolSize)
+        .description("The core number of threads.")
+        .tags(tags)
+        .register(meterRegistry);
+
+    Gauge.builder(metricName("tasks.completed.cumulative.count"), metrics::getCurrentCompletedTaskCount)
```

Comment:
Use `FunctionCounter` consider combining with `tasks.scheduled.cumulative.count` and using a tag

Reply:
Done.